### PR TITLE
pythonPackages.python-redis-lock: init at 3.3.1

### PR DESCRIPTION
**Pull request by vanschelven**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by risicle** [comment]
Note that though you've set `doCheck = true;`, it's not actually picking up any tests.

You may have to set the `checkPhase` to something like `pytest tests/` by the looks of it, of course adding `pytest` as a `checkInput`.

One trick you can use to decode how a project wants their tests to be run is following their `.travis.yml`.

**Comment by vanschelven** [comment]
> Note that though you've set `doCheck = true;`, it's not actually picking up any tests.

Thanks for pointing that out. The latest (squashed) push is running the tests

File: pkgs/development/python-modules/python-redis-lock/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  checkInputs = [ pytest process-tests pkgs.redis ];
+
+  checkPhase = ''
+    pytest tests/
+  '';
+
+  propagatedBuildInputs = [ redis ];
+
+  doCheck = true;
```
Comment (worldofpeace) [inline pkgs/development/python-modules/python-redis-lock/default.nix]:
```suggestion
```

Tests are default `true` for `buildPython*`

File: pkgs/development/python-modules/python-redis-lock/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/ionelmc/python-redis-lock";
+    license = licenses.bsd2;
+    description = "Lock context manager implemented via redis SETNX/BLPOP.";
```
Comment (worldofpeace) [inline pkgs/development/python-modules/python-redis-lock/default.nix]:
```suggestion
    description = "Lock context manager implemented via redis SETNX/BLPOP";
```

We don't use periods at the end of `description`.

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build python3Packages.python-redis-lock

File: pkgs/development/python-modules/python-redis-lock/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, redis
+, pytest
+, process-tests
+, pkgs
```
Comment (worldofpeace) [inline pkgs/development/python-modules/python-redis-lock/default.nix]:
```suggestion
, pkgs
, withDjango ? false, django_redis
```

There's an `extras_require` for this so we can add an argument to enable/disable the feature.

File: pkgs/development/python-modules/python-redis-lock/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    sha256 = "5316d473ce6ce86a774b9f9c110d84c3a9bd1a2abfda5d99e9c0c8a872a8e6d6";
+  };
+
+  checkInputs = [ pytest process-tests pkgs.redis ];
+
+  checkPhase = ''
+    pytest tests/
+  '';
+
+  propagatedBuildInputs = [ redis ];
```
Comment (worldofpeace) [inline pkgs/development/python-modules/python-redis-lock/default.nix]:
```suggestion
  propagatedBuildInputs = [ redis ] 
  ++ stdenv.lib.optional withDjango django_redis;
```

How you implement https://github.com/NixOS/nixpkgs/pull/62668/files#r291694999

**Comment by worldofpeace** [comment]
Thanks @vanschelven

**Comment by vanschelven** [comment]
That was quick... in fact too quick. I had only clicked the relevant buttons by @worldofpeace in preparation of doing some tests on my local machine, but I hadn't gotten the tests to run for the `withDjango = true` case yet.

In particular, I figured out the test-phase needs to be further adapted to read `DJANGO_SETTINGS_MODULE=test_project.settings pytest tests/`

However, that leaves one more broken test (which I haven't managed to debug yet)

```
/nix/store/qd45rwhc9nvqa5q3kkzhrf0v75j2yz9z-python3.7-django-redis-4.10.0/lib/python3.7/site-packages/django_redis/cache.py:32: in _decorator
    return method(self, *args, **kwargs)
/nix/store/qd45rwhc9nvqa5q3kkzhrf0v75j2yz9z-python3.7-django-redis-4.10.0/lib/python3.7/site-packages/django_redis/cache.py:81: in get
    client=client)
/nix/store/qd45rwhc9nvqa5q3kkzhrf0v75j2yz9z-python3.7-django-redis-4.10.0/lib/python3.7/site-packages/django_redis/client/default.py:205: in get
    raise ConnectionInterrupted(connection=client, parent=e)
E   django_redis.exceptions.ConnectionInterrupted: Redis ConnectionError: Error 32 while writing to socket. Broken pipe.

During handling of the above exception, another exception occurred:
tests/test_django_integration.py:28: in test_django_add_or_set_locked
    assert cache.locked_get_or_set("foobar-aosl", creator_42) == 42
/nix/store/cz2sgkfaxqgmv6xz7wq2yk7qyhw88lpm-python3.7-python-redis-lock-3.3.1/lib/python3.7/site-packages/redis_lock/django_cache.py:34: in locked_get_or_set
    val = self.get(key, version=version)
/nix/store/qd45rwhc9nvqa5q3kkzhrf0v75j2yz9z-python3.7-django-redis-4.10.0/lib/python3.7/site-packages/django_redis/cache.py:39: in _decorator
    raise e.parent
/nix/store/qd45rwhc9nvqa5q3kkzhrf0v75j2yz9z-python3.7-django-redis-4.10.0/lib/python3.7/site-packages/django_redis/client/default.py:203: in get
    value = client.get(key)
/nix/store/q922mzmsh9ldgsbp70kfn78wwcxskw1b-python3.7-redis-3.1.0/lib/python3.7/site-packages/redis/client.py:1264: in get
    return self.execute_command('GET', name)
/nix/store/q922mzmsh9ldgsbp70kfn78wwcxskw1b-python3.7-redis-3.1.0/lib/python3.7/site-packages/redis/client.py:774: in execute_command
    connection.send_command(*args)
/nix/store/q922mzmsh9ldgsbp70kfn78wwcxskw1b-python3.7-redis-3.1.0/lib/python3.7/site-packages/redis/connection.py:619: in send_command
    self.send_packed_command(self.pack_command(*args))
/nix/store/q922mzmsh9ldgsbp70kfn78wwcxskw1b-python3.7-redis-3.1.0/lib/python3.7/site-packages/redis/connection.py:612: in send_packed_command
```

**Comment by worldofpeace** [comment]
Perhaps you need to have `redis-server &`?

**Pull request merged**